### PR TITLE
Add NetApp E-Series module base class

### PR DESCRIPTION
**Pull request by ndswartz**
##### SUMMARY
Add NetApp E-Series module base class which provides common functionality for all future E-Series modules. Functionality provided is web services version, proxy, and URL checking along with an URL request method that wraps the current module_utils/netapp.py request function.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
NetAppESeriesModule class

##### ADDITIONAL INFORMATION
```config file = None
configured module search path = [u'/home/swartzn/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
ansible python module location = /home/swartzn/ansible-dev/lib/ansible
executable location = /home/swartzn/ansible-dev/bin/ansible
python version = 2.7.15rc1 (default, Nov 12 2018, 14:31:15) [GCC 7.3.0]
```

### CODE REVIEW

File: lib/ansible/module_utils/netapp.py [hunk 1]
```diff
@@ -45,6 +49,12 @@
     HAS_NETAPP_LIB = False
 
 import ssl
+import logging
```
Comment (gundalow) [inline lib/ansible/module_utils/netapp.py]: Please use Ansible's build in logging, see `module_utils/basic.py`

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/module_utils/netapp.py (hunk 1 above)
Comment (ndswartz) [reply lib/ansible/module_utils/netapp.py]: Thanks for letting me know about the built in Ansible logging. I went ahead and replaced the python logging with it.

**Review by lmprice — APPROVED** [review]

**Comment by ndswartz** [comment]
@gundalow I have multiple modules that are dependent on this change.  Should I go ahead an make pull requests for them, knowing that they will fail until this change is merged?

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
@ndswartz Merged, so you can now raise the remaining PRs.
Where ever possible we try to avoid PRs failing in CI due to dependent PRs

**Pull request merged**